### PR TITLE
[IoT Hub] Add more descriptive messages when exception raised

### PR DESCRIPTION
**Pull request by anusapan**
Return more descriptive message on raised exception from sdk.
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by anusapan** [comment]
@qwordy Could you review the PR?

**Comment by qwordy** [comment]
There are conflicting files

**Comment by yonzhan** [comment]
@haroldrandom please help with this PR.

**Comment by anusapan** [comment]
@haroldrandom Build passed successfully

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -70,6 +70,7 @@ Release History
 
 * Add new routing source type: DigitalTwinChangeEvents
 * Fix #2826: Missing features in "az iot hub create"
+* Bug Fixed: Return more descriptive message on raised exception.
```
Comment (haroldrandom) [inline src/azure-cli/HISTORY.rst]: Not **Bug fixed**, right? It's an enhancement. 
You could say like "{your subgroup}: Add more descriptive messages when exception raised."

Comment (haroldrandom) [reply src/azure-cli/HISTORY.rst]: The purpose of changing from HttpOperationError to CloudError is to get more information. The catching scope of HttpOperationError is too small. Which won't give the user any informative message.

**Pull request merged**